source/server-discovery-and-monitoring/server-monitoring.rst

```diff
@@ -667,10 +679,17 @@ The event API here is assumed to be like the standard `Python Event
                 # Wait before running the next check.
                 wait()
                 continue
-            topology.onServerDescriptionChanged(description)
-            if description.error != Null:
-                # Clear the connection pool only after the server description is set to Unknown.
-                clear connection pool for server
+
+            with client.lock:
+                # for drivers that implement CMAP, mark the connection pool as ready after
+                # performing a successful check.
+                if description.type != Unknown:
+                    mark connection pool for server as "ready"
+
+                topology.onServerDescriptionChanged(description)
+                if description.error != Null:
+                    # Clear the connection pool only after the server description is set to Unknown.
+                    clear connection pool for server
```

Comment:
We should make `onServerDescriptionChanged` call pool.ready(). The Monitor should not call this method directly because `onServerDescriptionChanged` deals with topolgyVersion checks and the Monitor does not. (see this comment: https://github.com/mongodb/specifications/pull/878/#discussion_r528914452)

Reply:
done